src/mime.cr

```diff
@@ -0,0 +1,167 @@
+require "crystal/system/mime"
+
+# This module implements a global MIME registry.
+#
+# ```
+# require "mime"
+#
+# MIME.fetch(".html") # => "text/html"
+# ```
+#
+# The registry is populated with some default values (see `DEFAULT_TYPES`)
+# as well as the operating system's MIME database.
+#
+# Applications can register their own MIME types:
+#
+# ```
+# MIME.fetch?(".cr")              # => nil
+# MIME.extensions("text/crystal") # => []
+#
+# MIME.register(".cr", "text/crystal")
+# MIME.fetch?(".cr")              # => "text/crystal"
+# MIME.extensions("text/crystal") # => [".cr"]
+# ```
+module MIME
+  class Error < Exception
+  end
+
+  @@initialized = false
+  @@types = {} of String => String
+  @@types_lower = {} of String => String
+  @@extensions = {} of String => Array(String)
+
+  # A limited set of default MIME types.
+  DEFAULT_TYPES = {
+    ".css"  => "text/css; charset=utf-8",
+    ".gif"  => "image/gif",
+    ".htm"  => "text/html; charset=utf-8",
+    ".html" => "text/html; charset=utf-8",
+    ".jpg"  => "image/jpeg",
+    ".jpeg" => "image/jpeg",
+    ".js"   => "application/x-javascript",
+    ".pdf"  => "application/pdf",
+    ".png"  => "image/png",
+    ".svg"  => "image/svg+xml",
+    ".txt"  => "text/plain; charset=utf-8",
+    ".xml"  => "text/xml; charset=utf-8",
+  }
+
+  private def self.initialize_types
+    return if @@initialized
+    @@initialized = true
+
+    DEFAULT_TYPES.each do |ext, type|
+      register ext, type
+    end
+
+    Crystal::System::MIME.load
+  end
+
+  # Looks up the MIME type associated with *extension*.
+  #
+  # A case sensitive search is tried first, if this yields not result, it it
+  # matched case-insensitive. Returns *default* if *extension* is not registered.
+  def self.fetch(extension : String, default) : String
```

Comment:
Perhaps rename `fetch` to `from_extension`? Or make `extension` required? `MIME.fetch(extension: ".foo")`. Having a `MIME.from_filename("index.html")` or a `MIME.fetch(filename: "index.html")` could be really nice (it just calls extname for you).

Reply:
Comment could be improved too: `if this yields not result, it it matched case-insensitive`.

Reply:
The problem with a required named argument is that the `default` argument would be weird. And I don't think named argument should be used to distinguish whether the method receives an extension or filename argument.

Reply:
`from_filename` and `from_extension` sounds good.

Although I really like `fetch` because it's short ;)